### DISCUSSION TITLE
[5.0][Feature request] Pass CrudPanel configuration to other field types

### DISCUSSION THREAD

**Discussion by tabacitu** [Laravel-Backpack/community-forum#340] [Ideas]
This _should_ be a good solution for both the "matrix field" #771, and for the "add items on the fly" #591 issue.

In order to create fields that take into consideration _what a connected Model needs to show in terms of columns/fields_, we can just pass its CrudController. It has all information that field will need:
- what columns to show
- what fields to show
- what the route is

So we should be able to do something like this:

```php
[ // CRUD Table as a Field Type - lists all related entries 
    'name' => 'categories',
    'label' => 'Categories',
    'type' => 'crud',
    'crud' => App\Http\Controllers\Admin\CategoryCrudController::class,
]
```

and

```php
[ // Select2 ajax with "add" button next to it, which opens a modal to add an entry
    'name' => 'categories',
    'label' => 'Categories',
    'type' => 'select2_from_crud',
    'crud' => App\Http\Controllers\Admin\CategoryCrudController::class,
    'attribute' => 'name',
]
```

**Comment by tswonke** [comment]
ah, never thought of that... interesting for my instant fields...
does this work already? did you try it?

**Comment by tabacitu** [comment]
Unfortunately, no, I haven’t tried it yet… :-( 
Still think it might be a good idea, though.

Up!

**Comment by tswonke** [comment]
I tried this last weekend in different ways, but I don't think it is possible to use it like this right now due to architectural reasons.

We would need to get access to a fully initialized instance of the CrudController to have knowledge about fields/columns etc. But this currently depends on the request itself (sometimes?).

Maybe(?) a static helper method could help out...
